sqlalchemy-stubs/orm/session.pyi

```diff
@@ -32,17 +33,274 @@ from ..sql.base import Options
+    @property
+    def dirty(self) -> util.IdentitySet[Any]: ...
+    @property
+    def deleted(self) -> util.IdentitySet[Any]: ...
+    @property
+    def new(self) -> util.IdentitySet[Any]: ...
+    @property
+    def is_active(self) -> bool: ...
+    @property
+    def no_autoflush(
+        self: _TSharedSessionProtocol,
+    ) -> ContextManager[_TSharedSessionProtocol]: ...
+    @util.memoized_property
+    def info(self) -> Mapping[Any, Any]: ...
+    def __contains__(self, instance: Any) -> bool: ...
+    def __iter__(self) -> Iterator[Any]: ...
+    def add(self, instance: Any, _warn: bool = ...) -> None: ...
+    def add_all(self, instances: Any) -> None: ...
+    def expire_all(self) -> None: ...
+    def expire(
+        self, instance: Any, attribute_names: Optional[Iterable[str]] = ...
+    ) -> None: ...
+    def expunge(self, instance: Any) -> None: ...
+    def expunge_all(self) -> None: ...
+    def get_bind(
+        self,
+        mapper: Optional[Any] = ...,
+        clause: Optional[ClauseElement] = ...,
+        bind: Optional[_T] = ...,
+        _sa_skip_events: Optional[Any] = ...,
+        _sa_skip_for_implicit_returning: bool = ...,
+    ) -> _T: ...
+    def is_modified(
+        self, instance: Any, include_collections: bool = ...
+    ) -> bool: ...
+    def in_transaction(self) -> bool: ...
+
+class _SessionProtocol(
+    _SharedSessionProtocol[Union[Connection, Engine]], Protocol
+):
+    def __enter__(self: _TSessionProtocol) -> _TSessionProtocol: ...
+    def __exit__(self, type_: Any, value: Any, traceback: Any) -> None: ...
+    def begin(
+        self,
+        subtransactions: bool = ...,  # NOTE: Deprecated.
+        nested: bool = ...,
+    ) -> _SessionTransactionProtocol: ...
+    def begin_nested(self) -> _SessionTransactionProtocol: ...
+    def rollback(self) -> None: ...
+    def commit(self) -> None: ...
+    def connection(self) -> Connection: ...
+    def execute(
+        self,
+        statement: Executable,
+        params: Optional[
+            Union[Mapping[str, Any], Sequence[Mapping[str, Any]]]
+        ] = ...,
+        execution_options: _ExecutionOptions = ...,
+        bind_arguments: Optional[_BindArguments] = ...,
+        **kw: Any,
+    ) -> Result: ...
+    def scalar(
+        self,
+        statement: Executable,
+        params: Optional[
+            Union[Mapping[str, Any], Sequence[Mapping[str, Any]]]
+        ] = ...,
+        execution_options: _ExecutionOptions = ...,
+        bind_arguments: Optional[_BindArguments] = ...,
+        **kw: Any,
+    ) -> Any: ...
+    def close(self) -> None: ...
+    def refresh(
+        self,
+        instance: Any,
+        attribute_names: Optional[Iterable[str]] = ...,
+        with_for_update: Optional[
+            Union[Literal[True], Mapping[str, Any]]
+        ] = ...,
+    ) -> None: ...
+    def delete(self, instance: Any) -> None: ...
+    def get(
+        self,
+        entity: Any,
+        ident: Any,
```

Comment:
I think this is could be `Union[Any, Tuple[Any, ...], Mapping[str, Any]`

Reply:
Any time you do a `Union[Any, ...]`, the net effect is `Any`

Reply:
sure, but it's a bit more informative I guess? in any case it's the same here. we can polish this later